components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java

```diff
@@ -2432,43 +2436,76 @@ private boolean addExternalAPIStoresDetails(APIIdentifier apiId,Set<APIStore> ap
+                    getAxisConfiguration().getClusteringAgent();
 
-            for (String gatewayEnvironmentName : gatewyEnvironments.keySet()) {
-                Environment currentGatewayEnvironment = gatewyEnvironments.get(gatewayEnvironmentName);
+            if (clusteringAgent != null) {
+                //changing stat publishing status at other nodes via a cluster message
+                try {
+                    clusteringAgent.sendMessage(new StatUpdateClusterMessage(updatedStatus), true);
+                } catch (ClusteringFault clusteringFault) {
+                    //error is only logged because initially gateway has modified the status
+                    log.error("Failed to send cluster message to Publisher/Store domain " +
+                            "and update stats publishing status.");
+                }
+                if (log.isDebugEnabled()) {
+                    log.debug("Successfully updated Stats publishing status to : " + updatedStatus);
+                }
+            }
+
+            Map<String, Environment> gatewayEnvironments = ServiceReferenceHolder.getInstance().
+                    getAPIManagerConfigurationService().getAPIManagerConfiguration().getApiGatewayEnvironments();
+
+            Set gatewayEntries = gatewayEnvironments.entrySet();
+            Iterator<Map.Entry<String,Environment>> gatewayIterator = gatewayEntries.iterator();
+
+            while(gatewayIterator.hasNext()){
+
+                Environment currentGatewayEnvironment = gatewayIterator.next().getValue();
                 String gatewayServiceUrl = currentGatewayEnvironment.getServerURL();
                 String gatewayUserName = currentGatewayEnvironment.getUserName();
                 String gatewayPassword = currentGatewayEnvironment.getPassword();
 
-                //get the stub and the call the admin service with the credentials
-                GatewayStatsUpdateServiceStub stub =
-                        new GatewayStatsUpdateServiceStub(gatewayServiceUrl + "GatewayStatsUpdateService");
-                StatUpdateStorePublisherDomain storePublisherMessageAgent = new StatUpdateStorePublisherDomain();
-                ServiceClient gatewayServiceClient = stub._getServiceClient();
-                CarbonUtils.setBasicAccessSecurityHeaders(gatewayUserName, gatewayPassword, gatewayServiceClient);
-
-                //send an empty string if at least one mandatory parameter is null
-                if (receiverUrl == null || user == null || password == null) {
-                    receiverUrl = "";
-                    user = "";
-                    password = "";
+                try {
+                    //get the stub and the call the admin service with the credentials
+                    GatewayStatsUpdateServiceStub stub =
+                            new GatewayStatsUpdateServiceStub(gatewayServiceUrl + "GatewayStatsUpdateService");
```

Comment:
Fixed in : https://github.com/wso2/carbon-apimgt/commit/67c064f7410655acdbfedac15735038bc54ddaa7